code_curation2_submissions2/Johanna_Transform_PRS.m

```diff
@@ -0,0 +1,150 @@
+PSN_PAT=PSN(~ind_HC);
+
+%% ACT 4: HC - Get auxiliary info for PRONIA table format
+% try once with sMRI auxiliary info and for remaining PSNs with IC_EC
+% questionnaire auxiliary info (this is due to the way PRONIA Portal
+% downloads are saved)
+
+% sMRI aux_info
+load(fullfile([aux_info_folder,'/sMRI/auxiliary_information_' visit 'HC_MRI_sMRI.mat']))
+[match_info_sMRI,match_PSN_sMRI]=nk_MatchID(aux_info_table.PATIENT_ID,aux_info_table,PSN_HC,PRS_HC);
+[PRS_notsMRI,PSN_notsMRI]=nk_MatchID(PSN_HC,PRS_HC,aux_info_table.PATIENT_ID,aux_info_table,'src_not_dst');
+% if not all PSNs accounted for, get info from IC_EC aux_info
+if ~isempty(PSN_notsMRI)
+    load(fullfile([aux_info_folder,'/Observer_Rating_Instruments/auxiliary_information_ScreeningHC_IC_EC.mat']));
+    [match_info_ICEC,match_PSN_ICEC]=nk_MatchID(aux_info_table.PATIENT_ID,aux_info_table,PSN_notsMRI,PRS_notsMRI);
+    [PRS_notICEC,PSN_notICEC]=nk_MatchID(PSN_notsMRI,PRS_notsMRI,aux_info_table.PATIENT_ID,aux_info_table,'src_not_dst');
+end
+
+% create empty table for PSNs that are not in the Reference tables but in
+% PRS table
+test=cell2table(cell(numel(PSN_notICEC),numel(aux_info_table.Properties.VariableNames)),'VariableNames',aux_info_table.Properties.VariableNames);
+test.PATIENT_ID=PSN_notICEC;
+test2=[test,PRS_notICEC];
+
+% Merge total table and save
+aux_total_table=[match_info_sMRI;match_info_ICEC;test2];
+aux_total=table2struct(aux_total_table);
+save(fullfile(out_folder,['complete_information_' visit 'HC_genetic_PRS.mat']),'aux_total_table','aux_total')
+
+
+%% ACT 5: PAT - Get auxiliary info for PRONIA table format
+
+% sMRI aux_info
+load(fullfile([aux_info_folder,'/sMRI/auxiliary_information_' visit 'PAT_MRI_sMRI.mat']))
+[match_info_sMRI,match_PSN_sMRI]=nk_MatchID(aux_info_table.PATIENT_ID,aux_info_table,PSN_PAT,PRS_PAT);
+[PRS_notsMRI,PSN_notsMRI]=nk_MatchID(PSN_PAT,PRS_PAT,aux_info_table.PATIENT_ID,aux_info_table,'src_not_dst');
+% if not all PSNs accounted for, get info from IC_EC aux_info
+if ~isempty(PSN_notsMRI)
+    load(fullfile([aux_info_folder,'/Observer_Rating_Instruments/auxiliary_information_ScreeningPAT_IC_EC.mat']));
+    [match_info_ICEC,match_PSN_ICEC]=nk_MatchID(aux_info_table.PATIENT_ID,aux_info_table,PSN_notsMRI,PRS_notsMRI);
+    [PRS_notICEC,PSN_notICEC]=nk_MatchID(PSN_notsMRI,PRS_notsMRI,aux_info_table.PATIENT_ID,aux_info_table,'src_not_dst');
+end
+
+% create empty table for PSNs that are not in the Reference tables but in
+% PRS table
+test=cell2table(cell(numel(PSN_notICEC),numel(aux_info_table.Properties.VariableNames)),'VariableNames',aux_info_table.Properties.VariableNames);
+test.PATIENT_ID=PSN_notICEC;
+test2=[test,PRS_notICEC];
+
+% Merge total table and save
+aux_total_table=[match_info_sMRI;match_info_ICEC;test2];
+aux_total=table2struct(aux_total_table);
+save(fullfile(out_folder,['complete_information_' visit 'PAT_genetic_PRS.mat']),'aux_total_table','aux_total')
+
+%% ACT 6: Output
+PRS_Tables=out_folder;
+end
```

Comment:
If there are constants, please declare Constants at the beginning of the script.
Explain them with comments.